spec/features/date_selection_spec.rb

```diff
@@ -0,0 +1,195 @@
+
+    expect_default_date_period_metrics_to_be_displayed(from, to)
+  end
+
+  it 'renders data for the last 30 days when `last 30 days` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 30.days
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-30-days').click
+    click_button 'OK'
+    sleep 2
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the previous month when `last month` is selected' do
+    to = Time.zone.today.last_month.end_of_month
+    from = Time.zone.today.last_month.beginning_of_month
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-month').click
+    click_button 'OK'
+    sleep 2
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the last 3 months when `last 3 months` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 3.months
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-3-months').click
+    click_button 'OK'
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the last 6 months when `last6 months` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 6.months
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-6-months').click
+    click_button 'OK'
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the last year when `last year` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 1.year
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-1-year').click
+    click_button 'OK'
+    sleep 2
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the last 2 years when `last 2 years` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 2.years
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-2-years').click
+    click_button 'OK'
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  def initial_page_stub
+    from = Time.zone.today - 30.days
+    to = Time.zone.today
+
+    content_data_api_has_metric(base_path: 'base/path',
+      from: from,
+      to: to,
+      metrics: %w[unique_pageviews pageviews number_of_internal_searches],
+      payload: {
```

Comment:
Do we need to be so explicit with the `payload`?
It should be enough with setting a few attributes (like `base_path`) and default the rest within `content_data_api_has_metric`?

Reply:
Sorry, if I wasn't clear enough. I meant having to be so explicit in the spec. We are not testing any of the metadata, so they shouldn't be part of the Spec. We could hide those attributes in the helper method, and override them with the values passed as parameters.